src/backend/commands/explain.c

```diff
@@ -541,11 +541,21 @@ ExplainOnePlan(PlannedStmt *plannedstmt, IntoClause *into, ExplainState *es,
 	else
 		dest = None_Receiver;
 
+	/*
+	 * When analyzing the query, we should avoid sending any DeclareCursorStmt
+	 * to the segments in order to select the correct execution strategy.
```

Comment:
Execution strategy is not so significant. In this case we should execute the select query on segments instead of opening cursor.